### PR TITLE
Single click to select + record

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
Fix a recently introduced issue with the countdown.

### CODE REVIEW

File: src/components/Gesture.svelte [hunk 1]
```diff
@@ -72,33 +71,36 @@
   let isThisRecording = false;
   let showCountdown = false;
   let countdownIdx = 0;
-  $: currCountdownConfig = countdownConfigs[countdownIdx];
```
Comment (microbit-grace) [inline src/components/Gesture.svelte:75]: Haven't quite understood why inlining `currCountdownConfig` fixes the bug (where the second time button B is pressed the recording doesn't work), but found the following: 

I tried adding `$: console.log("countdownIdx", countdownIdx)` before this line. Interestingly, `countdownIdx` doesn't get logged the second time button B gets pressed. 

But, if I move the reactive declaration(s) to [here](https://github.com/microbit-foundation/ml-trainer/blob/fea374478d5cb7af929e040a9c9eb63c04030cb7/src/components/Gesture.svelte#L265), which is after the `triggerButtonsClicked($buttonPressed)` is called, the `currCountdownConfig` reacts as expected, and the `countdownIdx` gets logged consistently when button B gets pressed.

It makes me think there is some magic going on regarding the ordering of these reactive statements. I tried to reproduce the issue via a playground script, but was unable to, which makes me think I don't understand this fully.

File: src/components/Gesture.svelte [hunk 2]
```diff
@@ -169,26 +171,8 @@
...
-  // If bluetooth is not connected, open connection prompt by calling callback
-  function selectClicked(): void {
-    if (!$state.isInputConnected) {
-      chosenGesture.update(gesture => {
-        gesture = null;
-        return gesture;
-      });
-      onNoMicrobitSelect();
-      return;
-    }
```
Comment (microbit-grace) [inline src/components/Gesture.svelte:183]: Removing `onNoMicrobitSelect` looks good to me! The record button is disabled when `!$state.isInputConnected`, so I don't think this part of the function can be ran at all. In short, `onNoMicrobitSelect` is not being called.

**Comment by microbit-grace** [comment]
When I try to use the screen reader to read out the new select and record aria label, I realise I couldn't do it because once focused, the row is automatically selected, so the record aria label is read out instead. Not sure if there is a way round it that would mean that it would actually get read out, but I guess it might be safer to have it?

File: src/components/Gesture.svelte [hunk 3]
```diff
@@ -72,33 +71,36 @@
...
+        if (!showCountdown) {
+          // recording cancelled
+          return;
+        }
+        if (countdownIdx < countdownConfigs.length) {
+          countdown(countdownConfigs[countdownIdx]);
+        } else {
+          recordData();
+          showCountdown = false;
+          // countdownIdx = 0;
```
Comment (microbit-matt-hillsdon) [inline src/components/Gesture.svelte:98]:
```suggestion
```

**Pull request merged**